### PR TITLE
create endpoint for create lesson by Teacher

### PR DESCRIPTION
**Pull request by MikolaLvivUA**

### CODE REVIEW

File: src/controllers/lesson/lesson.controller.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+      const lessonValidity = Joi.validate(lesson, lessonValidator);
+
+      if (lessonValidity.error) {
+        return next(new ErrorHandler(ResponseStatusCodesEnum.BAD_REQUEST, lessonValidity.error.details[0].message));
+      }
+
+      await lessonService.createLesson(lesson);
+
+      res.status(ResponseStatusCodesEnum.CREATED);
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts]: It's not will be working without .end

File: src/routes/api/api.router.ts [hunk 2]
```diff
@@ -1,17 +1,23 @@
 import { Router } from 'express';
+import { checkIsTeacher } from '../../middlewares/oauth';
```
Comment (VictorKmin) [inline src/routes/api/api.router.ts]: '../../middlewares';

File: src/routes/lesson/lesson.router.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Router } from 'express';
+
+import { lessonController } from '../../controllers';
+
+const router = Router();
+
+router.post('/', lessonController.createLesson);
```
Comment (VictorKmin) [inline src/routes/lesson/lesson.router.ts]: ckeckIsTeacher ?

File: src/services/lesson/lesson.service.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Lesson } from '../../database';
+import { ILesson } from '../../interfaces';
+
+class LessonService {
+
+  createLesson(lessonValue: ILesson): Promise<any> {
```
Comment (VictorKmin) [inline src/services/lesson/lesson.service.ts]: createLesson(lessonValue: ILesson): Promise<void> {

File: src/controllers/lesson/lesson.controller.ts (hunk 1 above)
Comment (MikolaLvivUA) [reply src/controllers/lesson/lesson.controller.ts]: fixed

**Pull request merged**